disperser/dataapi/metrics_handlers.go

```diff
@@ -142,3 +143,62 @@ func (s *server) getNonSigners(ctx context.Context, intervalSeconds int64) (*[]N
 
 	return &nonSignersObj, nil
 }
+
+func (s *server) getUnsignedBatches(ctx context.Context, intervalSeconds int64) (*OperatorNonsigningPercentage, error) {
+	nonSigners, err := s.subgraphClient.QueryBatchNonSigningOperatorIdsInInterval(ctx, intervalSeconds)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		limit        = 1000
+		skip         = 0
+		totalBatches = 0
+	)
+
+loop:
+	for {
+		select {
+		case <-ctx.Done():
+			break loop
+		default:
+			batches, err := s.subgraphClient.QueryBatchesWithLimit(ctx, limit, skip)
+			if err != nil {
+				s.logger.Error("Failed to query batches", "error", err)
+				return nil, err
+			}
+			totalBatches += len(batches)
+			if len(batches) == 0 {
+				break
+			}
+			skip += limit
+		}
+	}
+
+	percentage := make(map[string]float64, 0)
+
+	if totalBatches < 1 {
+		return &OperatorNonsigningPercentage{
+			TotalNonSigners:       len(nonSigners),
+			TotalBatches:          totalBatches,
+			PercentagePerOperator: percentage,
+		}, nil
+	}
+
+	for operatorId, totalUnsignedBatches := range nonSigners {
+		if totalUnsignedBatches > 0 {
+			ps := fmt.Sprintf("%.2f", float64(totalUnsignedBatches)/float64(totalBatches))
```

Comment:
what if there were a bunch of batches that they weren't an operator during? this number will be much higher than real. we need to find out how many batches they were active for that they signed. i recommend we refactor the subgraph if this is what we want

Reply:
to be more precise, the ratio calculated here is (total batches nonsigned by operator/total batches on eigenda ever), what we want is (total batches nonsigned by operator/total batches on eigenda that operator was requested to sign ever)

Reply:
Ok, that's good point. Then this number here will be lower than real, right? i.e. the numerator (total unsigned) is accurate, the denominator (the total batches) may be off, if the operator wasn't in network in last N seconds.